sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotNestTypesInArguments.cs

```diff
@@ -38,25 +38,29 @@ public sealed class DoNotNestTypesInArguments : SonarDiagnosticAnalyzer
         private const string DiagnosticId = "S4017";
         private const string MessageFormat = "Refactor this method to remove the nested type argument.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        private static readonly DiagnosticDescriptor rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
-            c =>
-            {
-                var argumentTypeSymbols = GetParametersSyntaxNodes(c.Node)
-                    .Where(p => GetNestingDepth(p, c) > 1);
-
-                foreach (var argument in argumentTypeSymbols)
+                c =>
                 {
-                    c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, argument.GetLocation()));
-                }
-            },
-            SyntaxKind.MethodDeclaration,
-            SyntaxKindEx.LocalFunctionStatement);
+                    var argumentTypeSymbols = GetParametersSyntaxNodes(c.Node).Where(p => MaxDepthReached(p, c.SemanticModel));
+
+                    foreach (var argument in argumentTypeSymbols)
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, argument.GetLocation()));
+                    }
+                },
+                SyntaxKind.MethodDeclaration,
+                SyntaxKindEx.LocalFunctionStatement);
+
+        private static bool MaxDepthReached(SyntaxNode parameterSyntax, SemanticModel model)
+        {
+            var walker = new GenericWalker(2, model);
+            walker.SafeVisit(parameterSyntax);
+            return walker.HasReachedMaxDepth;
```

Comment:
I suggest to give it the same name for consistency. Just to avoid confusion if the different wording has different meaning.
```suggestion
            return walker.MaxDepthReached;
```